include/CL/sycl/platform/detail/platform_tail.hpp

```diff
@@ -18,32 +18,35 @@ namespace sycl {
     @{
 */
 
-/** Return a list of all available devices
+/** Returns a vector class containing all SYCL devices associated with
+    this SYCL platform.
 
     Return synchronous errors via SYCL exception classes.
 */
 vector_class<device>
 platform::get_devices(info::device_type device_type) const {
-  // Start with the default device
-  vector_class<device> devices = { {} };
+  /** If \c get_devices is called with the host platform,
+      returns the host_device.
+  */
+  if(is_host()) return { {} };
```

Comment:
The problem is that there is a dependency between ```device``` and ```platform``` that forced me to declare ```platform::get_devices``` in *platform_tail.hpp*.